oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/version/VersionEditor.java

```diff
@@ -108,9 +112,15 @@ public void propertyAdded(PropertyState after)
                 && this.after.hasProperty(JcrConstants.JCR_VERSIONHISTORY)
                 && !this.after.hasProperty(JCR_ISCHECKEDOUT)
                 && !this.before.exists()) {
-            // sentinel node for restore
-            vMgr.restore(node, after.getValue(Type.REFERENCE), null);
-            return;
+            Tree tree = new TreeProviderService().createReadOnlyTree(this.node.getNodeState());
+            if (vMgr.getNodeTypeManager().isNodeType(
+                    TreeUtil.getPrimaryTypeName(tree), TreeUtil.getMixinTypeNames(tree), MIX_VERSIONABLE)) {
```

Comment:
Shouldn't this be replaced by `isVersionable()`. 
Both are using the same underlying apis and `isVersionable()` caches the results in `Boolean` as well.

Reply:
@rishabhdaim No, that doesn't work, because isVersionable() also checks for the presence of the jcr:isCheckedOut property.